keel-artifact/src/main/kotlin/com/netflix/spinnaker/keel/artifact/ArtifactListener.kt

```diff
@@ -54,22 +54,36 @@ class ArtifactListener(
     }
 
     if (artifactRepository.versions(artifact).isEmpty()) {
-      storeLatestVersion(artifact)
+      storeLatestVersion(artifact, event.statuses)
     }
   }
 
-  protected fun storeLatestVersion(artifact: DeliveryArtifact) =
+  /**
+   * Grab the latest version which matches the statuses we care about, so the artifact is relevant.
+   */
+  protected fun storeLatestVersion(artifact: DeliveryArtifact, statuses: List<ArtifactStatus>) =
     runBlocking {
       artifactService
-        .getVersions(artifact.name)
+        .getVersions(artifact.name, statuses)
         .firstOrNull()
         ?.let { firstVersion ->
           val version = "${artifact.name}-$firstVersion"
-          log.debug("Storing latest version {} for registered artifact {}", version, artifact)
-          artifactRepository.store(artifact, version)
+          val status = artifactStatus(artifactService.getArtifact(artifact.name, firstVersion))
+          log.debug("Storing latest version {} ({}) for registered artifact {}", version, status, artifact)
+          artifactRepository.store(artifact, version, status)
         }
     }
 
+  /**
+   * Parses the status from a kork artifact, and throws an error if [releaseStatus] isn't
+   * present in [metadata]
```

Comment:
This might be an assumption we have to re-work later, but I think it's ok for now.